src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -62,7 +62,7 @@ public bool IsSingleSegment
         public SequencePosition Start
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => new SequencePosition(_startObject, _startInteger);
+            get => new SequencePosition(_startObject, _startInteger & ReadOnlySequence.IndexBitMask);
```

Comment:
we can reuse the helper method that already takes care of that and is used in other places in the source code:

https://github.com/dotnet/runtime/blob/ec1897724f6dc5ab49eebde5f35292493e1bed7b/src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.Helpers.cs#L467-L468

```suggestion
            get => new SequencePosition(_startObject, GetIndex(_startInteger));
```